app/src/main/java/ui/transcript/TranscriptActivity.kt

```diff
@@ -94,7 +97,7 @@ class TranscriptActivity : DrawerActivity() {
             return
         }
 
-        launch {
+        launch(uiDispatcher) {
```

Comment:
Isn't it ui dispatcher by default? As a note, if you do things like this a lot, you can use conflated `BroadcastChannel` to support backpressure. That way if you have multiple pending refreshes, it will only be done once